From 501c48715a93b7e47d043261d5830dceb458c053 Mon Sep 17 00:00:00 2001
From: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
Date: Thu, 14 Dec 2017 13:19:38 +0530
Subject: [PATCH 0992/1240] Soundwire: Fix out of bounds access to
 sdw_num_row_mapping

In sdw_get_row_to_num() the sdw_num_row_mapping can be accessed at
MAX_NUM_ROWS index and will lead to out of bound access.

Fix out of bound access by correcting the for loop condition.

Change-Id: Iefeb91a2ab9e2bbf8376441562c3a19898cebe57
Review-by: Vaibhav Agarwal <vaibhav.agarwal@intel.com>
Signed-off-by: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
---
 drivers/sdw/sdw_bwcalc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/sdw/sdw_bwcalc.c b/drivers/sdw/sdw_bwcalc.c
index 57dc85d..a620942 100644
--- a/drivers/sdw/sdw_bwcalc.c
+++ b/drivers/sdw/sdw_bwcalc.c
@@ -152,7 +152,7 @@ int sdw_get_row_to_num(int row)
 {
 	int i;
 
-	for (i = 0; i <= MAX_NUM_ROWS; i++) {
+	for (i = 0; i < MAX_NUM_ROWS; i++) {
 		if (sdw_num_row_mapping[i].row == row)
 			return sdw_num_row_mapping[i].num;
 	}
-- 
2.7.4

